File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;


import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌服务层接口
 */

public interface BrandService {
    //返回全部列表
    public List<TbBrand> findAll();


    //返回分页列表
    public PageResult findPage(int pageNum, int pageSize);

    //增加品牌
    public void addBrand(TbBrand brand);

    //更新品牌
    public void updateBrand(TbBrand brand);

    //根据id查找品牌
    public TbBrand findOne(Long id);

    //批量删除选中的ids(集合)的记录
    public void deleteBrand(Long[] ids);


    PageResult findPage(TbBrand tbBrand, int pageNum, int pageRow);


    /**
     * 品牌下拉框数据
     */
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌 服务层
app.service("brandService",function ($http) {

    //读取列表数据绑定到表单中
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }

    this.findPage=function (page,size) {
        return $http.get("../brand/findPage.do?page="+page+"&size="+size);
    }

    this.add=function (entity) {
        return $http.post("../brand/add.do",entity);
    }

    this.updateBrand=function (entity) {
        return $http.post("../brand/updateBrand.do",entity);
    }

    this.delet=function (ids) {
        return $http.get("../brand/deleteBrand.do?ids="+ids);
    }

    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }

    this.search=function (page,size,searchEntity) {
        return $http.post("../brand/searchPage.do?page="+page+"&size="+size,searchEntity);
    }


    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }
});